Release v0.21.0

seethroughdev opened the pull request:
This release addresses: https://github.com/seethroughdev/obsidian-recipe-grabber/issues/32

seethroughdev commented:
Just assigning review in case anyone is open to it. Otherwise, i'll merge it later. Thanks!

seethroughdev commented on src/modal-load-recipe.ts:
```diff
@@ -16,6 +23,13 @@ export class LoadRecipeModal extends Modal {
 
 		new Setting(contentEl).setName("url: ").addText((text) => {
 			text.setPlaceholder("https://www.example.com/recipe");
+
```
Added this back in to just auto-add a recipe url for development and testing. If this is a bad idea, happy to pull it back out..

Ovyerus replied:
It's probably better not to have this, otherwise it makes it impossible to turn on debug mode to see errors being thrown with a particular URL (e.g. like what happened for #31). Or at least have it a editable text field setting to be able to test with different URLs as needed.

seethroughdev replied:
Thinking the same thing. I’ll either remove, comment it out or make it a setting like you suggested. Thanks!

seethroughdev commented on src/main.ts:
```diff
@@ -486,19 +489,22 @@ export default class RecipeGrabber extends Plugin {
 		const traverse = (obj: any) => {
+			if (!obj) return;
+
 			Object.keys(obj).forEach((key) => {
```
assigned to value just to make it easier to follow.

seethroughdev replied:
And added escape because for some values, the recursion was throwing an error.

seethroughdev commented:
Thinking the same thing. I’ll either remove, comment it out or make it a setting like you suggested. Thanks!On May 25, 2024, at 5:44 PM, Michael Mitchell ***@***.***> wrote:﻿
@Ovyerus commented on this pull request.



In src/modal-load-recipe.ts:
> @@ -16,6 +23,13 @@ export class LoadRecipeModal extends Modal {
 
 		new Setting(contentEl).setName("url: ").addText((text) => {
 			text.setPlaceholder("https://www.example.com/recipe");
+

It's probably better not to have this, otherwise it makes it impossible to turn on debug mode to see errors being thrown with a particular URL (e.g. like what happened for #31). Or at least have it a editable text field setting to be able to test with different URLs as needed.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

seethroughdev commented:
Okay, I cleaned up the previous PR, couple of notes:

- removed entire recursive unescape.
- replaced with: https://github.com/seethroughdev/obsidian-recipe-grabber/pull/20/files#diff-4fab5baaca5c14d2de62d8d2fceef376ddddcc8e9509d86cfa5643f51b89ce3dR166
- basically removed all escaping issues, as i think that can be well handled with handlebars templates instead
- replaced with a `decodeEntities` option that you can disable if you are having utf8 issues

Also, hoping to make the formatting standard so we don't have to worry about it anymore. I added explicit prettier rules that cleaned up all the files. If you are looking at the diff, you might want to disable whitespace in the UI, or clicking https://github.com/seethroughdev/obsidian-recipe-grabber/pull/33/files?w=1

Thanks again!

seethroughdev commented:
Thanks @Ovyerus & @Flxp49 !
